core/vm/evm.go

```diff
@@ -381,7 +400,7 @@ func (evm *EVM) StaticCall(caller ContractRef, addr common.Address, input []byte
 	}
 
 	if p, isPrecompile := evm.precompile(addr); isPrecompile {
-		ret, gas, err = RunPrecompiledContract(p, input, gas)
+		ret, gas, err = RunStatefulPrecompiledContract(p, evm, caller.Address(), addr, input, gas, evm.interpreter.readOnly)
```

Comment:
I'm not 100% we should be passing readOnly var for static call -- is this always true or is it only true run in non-tx contexts?  Need to ensure an executed precompile is able to infer & panic if tx requests a state changing call in staticcall

Reply:
Fixed:
- https://github.com/Kava-Labs/go-ethereum/pull/10/commits/74993c4c1ed5f7ee9edbfb4513b6707a471f752b
- https://github.com/Kava-Labs/go-ethereum/pull/10/commits/e08309004de5e54f20c67549ada19f6133a66706
- https://github.com/Kava-Labs/go-ethereum/pull/10/commits/57724b99e87b5072a28e07fcd18cca97ff8e5d5b